lib/roby/app.rb

```diff
@@ -399,6 +399,29 @@ def search_path
             end
         end
 
+        def register_app(path)
+            path = File.expand_path(path, app_dir)
+            app_name = File.basename(path)
+            @search_path ||= self.search_path
```

Comment:
```suggestion
            @search_path |= self.search_path
```
Is it set addition?

Reply:
Actually not ... `self.search_path` will return the current search path, which might be different than `@search_path`. `||=` is "assigned if currently false or nil`

To be honest, I think that one good thing to do would be to change the code to simply set `@search_path`. But since I'm not too confident about other effects this could have, I elected to keep the current code as-is and do this here.